tests/js/client/shell/api/multi/coordinator-metrics-cluster.js

```diff
@@ -110,7 +111,20 @@ function checkCoordinators(coordinators, mode) {
   assertTrue(coordinators.length > 1);
   for (let i = 1; i < coordinators.length; i++) {
     let c = coordinators[i];
-    let txt = getAllMetric(c, mode);
+    let txt;
+    let numSecs = 0.5;
+    // this loop is for checking that the document metrics from iresearch are present in the metrics, otherwise, we retry
+    // or else, in the checkRawMetrics() function, the parsing of the document fields in the metrics object would return
+    // undefined
+    while (true) {
+      txt = getAllMetric(c, mode);
+      if (txt.includes("arangodb_search_num_docs") || numSecs >= 20) {
+        break;
+      }
+      internal.sleep(numSecs);
+      numSecs *= 2;
+    }
+    assertTrue(txt.includes("arangodb_search_num_docs"), "couldn't acquire document related metrics info");
     checkRawMetrics(txt, false);
   }
```

Comment:
Oh, I see now, Alexey some time before fix it incorrect way.
Do you see checkIndexMetrics above? it's called for constant value.
Instead of your code you need to call checkIndexMetrics and remove checkIndexMetrics above